diff --git a/egui-d3d9/src/app.rs b/egui-d3d9/src/app.rs
index 0050973..166f032 100644
--- a/egui-d3d9/src/app.rs
+++ b/egui-d3d9/src/app.rs
@@ -81,6 +81,7 @@ impl<T> EguiDx9<T> {
         if self.should_reset {
             self.buffers = Buffers::create_buffers(dev, 16384, 16384);
             self.tex_man.reallocate_textures(dev);
+            self.ctx = egui::Context::default();
         }
 
         let output = self.ctx.run(self.input_man.collect_input(), |ctx| {
